site/source/docs/configuration.md

```diff
@@ -35,7 +35,25 @@ Every application [created with Embark](create_project.html) comes with an `emba
 }
 ```
 
-Let's look at the different options and learn what they do and mean.
+Alternatively, it's also possible to use a `embark.config.js` file instead, which exports a function that calculates the embark configuration object. This is particularly useful when the configuration needs to be built based on asynchronous operations. To make use of `embark.config.js`, simply create the file in your DApp and make sure it exports an (async) function which resolves with a config object like this:
+
+```js
+// embark.config.js
+
+module.exports = async function () {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve({
+        // configuration goes here
+      })
+    }, 5000);
+  });
+};
+```
+
+Embark will import and run the exported functoin inside an `async/await` context. So it's also possible to simply return the configuration object without wrapping it inside a `Promise`. Notice that the `setTimeout()` in the snippet above is just for demonstrating asynchronisity.
```

Comment:
```suggestion
Embark will import and run the exported function inside an `async/await` context. So it's also possible to simply return the configuration object without wrapping it inside a `Promise`. Notice that the `setTimeout()` in the snippet above is just for demonstrating asynchronicity.
```

Reply:
If you change the snippet like I suggested, this whole paragraph will need to be changed